client/src/unifycr-dirops.c

```diff
@@ -115,78 +143,124 @@ DIR *UNIFYCR_WRAP(opendir)(const char *name)
         }
 
         meta = unifycr_get_meta_from_fid(fid);
-        meta->is_dir = 1;
-        meta->size = sb->st_size;
-        meta->chunks = sb->st_blocks;
+        meta->is_dir   = 1;
+        meta->size     = sb->st_size;
+        meta->chunks   = sb->st_blocks;
         meta->log_size = 0;
     }
 
-    _dirp = unifycr_dirstream_alloc(fid);
+    unifycr_dirstream_t *dirp = unifycr_dirstream_alloc(fid);
 
-    return (DIR *) _dirp;
+    return (DIR *) dirp;
 }
 
 DIR *UNIFYCR_WRAP(fdopendir)(int fd)
 {
-    fprintf(stderr, "Function not yet supported @ %s:%d\n", __FILE__, __LINE__);
-    errno = ENOSYS;
-
-    return NULL;
+    if (unifycr_intercept_fd(&fd)) {
+        fprintf(stderr, "Function not yet supported @ %s:%d\n",
+            __FILE__, __LINE__);
+        errno = ENOSYS;
+        return NULL;
+    } else {
+        MAP_OR_FAIL(fdopendir);
+        DIR *ret = UNIFYCR_REAL(fdopendir)(fd);
+        return ret;
+    }
 }
 
 int UNIFYCR_WRAP(closedir)(DIR *dirp)
 {
-    fprintf(stderr, "Function not yet supported @ %s:%d\n", __FILE__, __LINE__);
-    errno = ENOSYS;
-
-    return -1;
+    if (unifycr_intercept_dirstream(dirp)) {
```

Comment:
Don't have any tests for these dirops wrappers yet (granted most of them are "not yet supported"). However, as they get implemented, we may end up needing to check if `dirp` is a valid value as calls like `closedir` can return `EBADF` and others like `dirfd`can return `EINVAL`.

Reply:
Good point.  The intercept should return if the address the user gives is in range.  We should double check that the structure is still in use after that.